### DISCUSSION TITLE
Finding sprite/element by name

### DISCUSSION THREAD

**Discussion by k11q** [pixijs/pixijs#9005] [Q&A] [answered]
Hi how can I find my element directly? I have many nested elements and I identify each of them with unique ID for 'name', I can loop through recursively, but is there another way do directly find them without having to loop? Thank you

**Comment by bigtimebuddy** [comment, marked as answer]
You can use `getChildByName`. The first parameter is the name you're looking for, the second optional parameter here if you should search the whole tree. 

```js
const app = new PIXI.Application();
app.stage.getChildByName('myName');
```

This method, however, is looping through all the elements recursively. Alternatively you could create and manage a `Map` of your elements for the quickest possible lookup. Pixi doesn't create this data-structure for you so the lookup is slow on very large searches.

**Reply by k11q** [reply]
I see. I think that is very inconvenient. I mean html for example has getelementbyid which i think is big o of 1 right? What is the best method to approach this, mapping?

**Reply by bigtimebuddy** [reply]
I don't think there's a right answer. You could do something maybe like this:

**utils.js**

```js

const elementMap = new Map();

export function createElement(ClassType, id) {
  const obj = new ClassType();
  obj.once('destroyed', () => elementMap.delete(id));
  elementMap.set(id, obj);
  return obj;
}

export function getElementById(id) {
  return elementMap.get(id) ?? null;
}
```

**index.js**

```js
import { Sprite } from 'pixi.js';
import { createElement, getElementById } from './utils';

const sprite = createElement(Sprite, 'my-sprite');
const sprite2 = getElementById('my-sprite');

console.assert(sprite === sprite2);
```

**Reply by k11q** [reply]
Ahh that's silly, sorry, I am new to javascript and I only knew about array.map, I thought that's what you meant. Thank you for the great explanation and example!
